ytmusicapi/mixins/browsing.py

```diff
@@ -75,41 +98,60 @@ def search(self, query: str, filter: str = None) -> List[Dict]:
                     param2 = 'RAAGAAgACgA'
                 params = param1 + param2 + param3
 
-            body['params'] = params
-
-        response = self._send_request(endpoint, body)
-
-        try:
-            # no results
-            if 'contents' not in response:
-                return search_results
+        type = filter[:-1] if filter else None
 
-            if 'tabbedSearchResultsRenderer' in response['contents']:
-                results = response['contents']['tabbedSearchResultsRenderer']['tabs'][int(
-                    filter == "uploads")]['tabRenderer']['content']
-            else:
-                results = response['contents']
+        if not continuationToken:
+            body = {"query": q, "params": params}
 
-            results = nav(results, SECTION_LIST)
+        if continuationToken:
+            query = {"cToken": continuationToken, "continuation": continuationToken}
 
-            # no results
-            if len(results) == 1 and 'itemSectionRenderer' in results:
-                return search_results
+        response = self._send_request(endpoint, body, query=query)
 
-            for res in results:
-                if 'musicShelfRenderer' in res:
-                    results = res['musicShelfRenderer']['contents']
+        try:
+            results = None
+
+            if "continuationContents" in response:
+                results = response["continuationContents"]
+            elif "contents" in response:
+                if "tabbedSearchResultsRenderer" in response["contents"]:
+                    results = response["contents"]["tabbedSearchResultsRenderer"]["tabs"][int(
+                        filter ==  # noqa: W504
+                        "uploads")]["tabRenderer"]["content"]["sectionListRenderer"]["contents"]
+                else:
+                    results = response["contents"]["sectionListRenderer"]["contents"]
+
+            if not results:
+                return
+
+            items = []
+            nextToken = None
+            if "musicShelfContinuation" in results:
+                musicShelf = results["musicShelfContinuation"]
+                items.append(musicShelf["contents"])
+                if "continuations" in musicShelf:
+                    nextToken = musicShelf["continuations"][0]["nextContinuationData"][
+                        "continuation"]
+            else:
+                for content in results:
+                    musicShelf = content["musicShelfRenderer"]
+                    items.append(musicShelf["contents"])
+                    if musicShelf["continuations"]:
+                        nextToken = musicShelf["continuations"][0]["nextContinuationData"][
+                            "continuation"]
```

Comment:
This part will probably not be needed when you use get_continuations

Reply:
Noted.